<launch>
<include file="$(find transporter)/launch/no_ekf.launch" />

  <node pkg="hector_imu_attitude_to_tf" type="imu_attitude_to_tf_node" name="imu_attitude_to_tf_node" output="screen">
    <remap from="imu_topic" to="imu/data" />
    <param name="base_stabilized_frame" type="string" value="base_stabilized" />
    <param name="base_frame" type="string" value="base_link" />
  </node>

  <!--- Run Move Base (to get path planning params)
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <rosparam file="$(find transporter_navigation)/explore/costmap_common.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find transporter_navigation)/explore/global_costmap.yaml" command="load" />
    <rosparam file="$(find transporter_navigation)/explore/navfn_params.yaml" command="load" />
    
    <rosparam file="$(find transporter_navigation)/explore/costmap_common.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find transporter_navigation)/explore/local_costmap.yaml" command="load" />
    <rosparam file="$(find transporter_navigation)/explore/trajectory_planner_params.yaml" command="load" />
  </node>  -->

  <!--- Run hector mapping -->
  <node pkg="hector_mapping" type="hector_mapping" name="hector_mapping" output="screen">
    <remap from="map" to="hectormap" />
    <remap from="map_metadata" to="hectormap_metadata" />
    
    <param name="pub_map_odom_transform" value="true"/>
    <param name="pub_map_scanmatch_transform" value="false"/>
    <param name="map_frame" value="odom" />
    <param name="base_frame" value="base_stabilized" />
    <param name="odom_frame" value="base_stabilized" />

    <!-- Map size / start point -->
    <param name="map_resolution" value="0.050"/>
    <param name="map_size" value="2048"/>
    <param name="map_start_x" value="0.5"/>
    <param name="map_start_y" value="0.5" />
    <param name="map_multi_res_levels" value="3" />
    
    <!-- Map update parameters -->
    <param name="update_factor_free" value="0.4"/>
    <param name="update_factor_occupied" value="0.95" />    
    <param name="map_update_distance_thresh" value="0.3"/>
    <param name="map_update_angle_thresh" value="0.17" />
    <param name="map_pub_period" value="2.0"/>

    <param name="laser_min_dist" value="0.03" />
    <param name="laser_max_dist" value="30.0" />
    <param name="laser_z_min_value" value="-1.95"/>
    <param name="laser_z_max_value" value="1.95"/>

  </node>
 
</launch>
